### DISCUSSION TITLE
Getting 404 error when everything else seems to be fine

### DISCUSSION THREAD

**Discussion by miguel-ossa** [vercel/community#4960] [Help] [answered]
### Summary

I get the error after pushing a python app with flask. The deployment is ok and the contents of the directory are also fine. My main file is main.py and this is the content of my package.json:

`{
    "version": "2",
    "builds": [
        {
            "src": "*.py",
            "use": "@miguel-ossa/now-flask"
        }
    ],
    "routes": [
        {
            "src": "(.*)",
            "dest": "main.py"
        }
    ]
  }`

![vercel](https://github.com/vercel/community/assets/9202841/c34f97ba-6d00-4197-a313-0dfdfca79243)

### Example

https://vercel.com/miguel-ossas-projects/python-blog/Cphwum9BhBHotFRdDz1wCujMnPx8

### Steps to Reproduce

You can visit:

[https://python-blog-psi.vercel.app/](url)

**Comment by amyegan** [comment]
Hey @miguel-ossa. Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that we can use to help with debugging?

https://vercel.com/docs/functions/serverless-functions/runtimes/python

**Reply by miguel-ossa** [reply]
The main.py has nearly 300 lines, with another module nearly 35 lines. The code works fine on local.

And here is the code on github:

[https://github.com/miguel-ossa/python-blog](https://github.com/miguel-ossa/python-blog)

**Reply by amyegan** [reply, marked as answer]
I think your repo is private. I get a 404 when I try to reach it. 

We do have some [Flask templates](https://vercel.com/templates?search=flask) available which might be helpful for reference. I have run into problems with some specific versions though. For example, I've seen problems with `2.2.2` and `2.2.3` recently. But I just deployed a new project from [Flask Hello World](https://vercel.com/templates/python/flask-hello-world) template on version `3.0.0` and it's working well for me.

**Reply by miguel-ossa** [reply]
No, is public:, but the link didn't work, now it is solved.

[https://github.com/miguel-ossa/python-blog](https://github.com/miguel-ossa/python-blog)

![Captura desde 2023-11-25 08-18-21](https://github.com/vercel/community/assets/9202841/53463fb2-414a-411e-a1f2-957817bd9740)

But I will create another one, and I will go step by step, trying with your suggestions. Thanks.

**Reply by miguel-ossa** [reply]
Ok, finally I see that the main python file must be in a folder inside the solution, not in the root of the project. This is a weird restriction... anyway, I will continue doing in this line.

[https://github.com/miguel-ossa/tresMercadosBlog.git](https://github.com/miguel-ossa/tresMercadosBlog.git)

[https://tres-mercados-blog.vercel.app/](https://tres-mercados-blog.vercel.app/)

**Reply by miguel-ossa** [reply]
Everything is solved. I am working with:

`Flask==2.2.5`

[https://tres-mercados-blog.vercel.app/](https://tres-mercados-blog.vercel.app/)

[https://github.com/miguel-ossa/tresMercadosBlog.git](https://github.com/miguel-ossa/tresMercadosBlog.git)
